.github/workflows/instascale-release.yml

```diff
@@ -1,37 +1,66 @@
-# This workflow will build an image for the instascale-controller, then push the image to the `project-codeflare` quay repo
+# This workflow will release the InstaScale repository: build an image for the instascale-controller, then push the image to the `project-codeflare` quay repo
 
 name: InstaScale Release
 
 on:
   workflow_dispatch:
-  push:
-    branches: [ "main" ]
-    paths:
-      - "VERSION"
+    inputs:
+      tag:
+        description: 'Release tag'
+        required: true
+        default: 'v0.0.0-dev'
+      mcad-version:
+        description: 'Published version of multi-cluster-app-dispatcher to be used as a dependency for InstaScale'
+        required: true
+        default: 'v0.0.0-dev'
 
 jobs:
   release:
     runs-on: ubuntu-latest
+
+    # Permission required to create a release
+    permissions:
+      contents: write
+
     steps:
     - uses: actions/checkout@v3
 
-    - name: Grab New Version Tag
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+
+    - name: Update to specific version of multi-cluster-app-dispatcher dependency
       run: |
-        echo "tag=$(cat VERSION)" >> $GITHUB_ENV
+        go get github.com/project-codeflare/multi-cluster-app-dispatcher@${{ github.event.inputs.mcad-version }}
 
+    # Build and test repository to make sure that MCAD update didn't break anything
+    - name: Build
+      run: go build -o bin/manager main.go
+    - name: Test
+      run: go test -v ./...
+
+    # Build and push the image
     - name: Login to Quay.io
       uses: redhat-actions/podman-login@v1
       with:
         username: ${{ secrets.QUAY_ID }}
         password: ${{ secrets.QUAY_TOKEN }}
         registry: quay.io
-
     - name: Image Build
       run: |
-        make image-build -e IMG=quay.io/project-codeflare/instascale-controller:v${{ env.tag }}
-        podman tag quay.io/project-codeflare/instascale-controller:v${{ env.tag }} quay.io/project-codeflare/instascale-controller:latest
-
+        make image-build -e IMG=quay.io/project-codeflare/instascale-controller:${{ github.event.inputs.tag }}
+        podman tag quay.io/project-codeflare/instascale-controller:${{ github.event.inputs.tag }} quay.io/project-codeflare/instascale-controller:latest
```

Comment:
I think the `latest` tag should only apply to the latest version, i.e., based on the semantic versioning ordering, not the latest performed release.

Reply:
hmm, good point
maybe there should be a boolean flag added as part of manual invocation indicating that this is a latest version (disabled by default to prevent accidental latest tag)?

Reply:
Done, added a checkbox to tag and push `latest` tag, disabled by default.